app/helpers/argo_helper.rb

```diff
@@ -179,7 +179,7 @@ def render_foxml_link(document, link_text = 'FoXML', opts = {:target => '_blank'
   end
 
   def render_solr_link(document)
-    return '' unless current_user.is_admin
+    return '' unless can? :read, document
```

Comment:
It's exactly the same. One of the cancan rules is "admins can do anything":

> can :manage, :all if user.is_admin?

And there are no permissions for other users.